pkg/jobs/jobs.go

```diff
@@ -160,16 +156,16 @@ func (j *Job) Dispatch(ctx context.Context, tasks []interface{}) *WorkerError {
 				if stoppedWorkersCount == 1 {
 					// at least one worker exited - we are done
 					// Unlock all others waiting to get a task
-					if stopped := j.Queue.Stop(); stopped {
-						// klog.V(6).Infof("Workqueue stopped\n")
+					if j.Queue.Stop() {
+						klog.V(1).Infof("Workqueue stopped\n")
 					}
 				}
 				if stoppedWorkersCount == workersCount {
-					var s string
 					if j.Queue.Count() > 0 {
-						s = fmt.Sprintf("%d unprocessed items in queue. ", j.Queue.Count())
+						klog.Warningf(
+							fmt.Sprintf("%d unprocessed items in queue. ", j.Queue.Count()),
+						)
```

Comment:
Ideally, the type of log (level) will depend on the situation. If an error that occurred in a worker of a job configured for fail-fast having some unprocessed items in the queue is expected situation, i.e. the log is simply informative. If other cases, that's most likely a program fault (bug), because except for fail-fast jobs the queues are gracefully drained before stopping the controllers.
Warning is a safe option in most cases. A future improvement could be to account in a more fine-grained manner for the situation discussed above and raise and error accordingly.